rep-2009.rst

```diff
@@ -0,0 +1,465 @@
+
+.. code-block::
+
+    N_1* := N_1([x, y, z], {x: [x, y, z], y: [y, z, x], z: [z, x, y]})
+    N_2* := N_1*
+    N_3* := N_1*
+
+   (5c)
+
+       -- N_3* <--
+       |         |
+       v         |
+     N_1* ----> N_2*
+
+Notice, however, that the deadlock is fixed by one node readily revealing its preferences.
+
+.. code-block::
+
+    N_1* := N_1([x, y, z], {x: [x, y, z], y: [y, z, x], z: [z, x, y]})
+    N_2* := N_1*
+
+   (5c)
+
+       -- N_3(x, y, z) <--
+     x |                 |
+       v    x          x |
+     N_1* ----> N_2* -----
+
+
+Rationale
+=========
+
+Having the Publisher Pick the Message Type
+------------------------------------------
+
+Consider a network with ``m`` publishers and ``n`` subscribers, where ``m`` and ``n`` are positive integers.
+Also imagine that there are at least two publishers that are publishing on the same topic.
+In this case, it is possible to have each of these publishers consider the other publishers in their decision of what message type to send.
+
+It is also true that loops in the network may occur.
+For example, imagine nodes ``A``, ``B``, and ``C``.
+``A`` sends a message to ``B``, and ``B`` sends a message to ``C``.
+This gets more complicated if ``A`` also sends a message to ``C``.
+
+In both of the above cases, it is much more challenging to find the best message type than the simple strategy detailed in a previous section.
+It was thought that the simpler approach described above in the specifications gets us almost all the way there, while being much simpler to implement.
+In addition, if it turns out to be necessary, the simpler approach can always be replaced by a better method for getting the optimal solution in future work.
+
+
+Using Different Topics or the Same Topic For Different Message Types
+--------------------------------------------------------------------
+
+If a publisher can publish ``ARGB888`` or ``YUV420`` images, should the images be published on namespaced topics or to the topic directly?
+Specifically, say that it has been negotiated the the publisher should publish ``ARG888`` on the topic ``/image``, should the images be published on ``/image/ARGB888`` or ``/image``?
+Using ``/image/ARGB888`` removes ambiguity, but could make it hard to work with other nodes that do not negotiate their message types that are also on the network.
+While ``/image`` is cleaner, but may lead to confusion when a single topic is used for multiple message types.
+``/image`` also makes it more difficult for the user to use topics published by nodes that do not perform type negotiation, which actually may be desirable: it will reduce the chance that the user builds a pipeline for one message type but then receives another message type on the same topic.
+
+Since the ``<topic>/<message type>`` format removes ambiguity and makes it harder to use negotiated nodes inappropriately, we will use the ``<topic>/<message type>`` format for all topics.
```

Comment:
I'd make sure to mention here that we'll still apply the remapping rules, particularly namespaces.  The true remapping rules (`--ros-args -r foo:=bar`) will be harder to use in the negotiated way, but should still be possible.